pkg/querysharding/analyzer.go

```diff
@@ -20,12 +25,41 @@ var nonShardableFuncs = []string{
 
 // NewQueryAnalyzer creates a new QueryAnalyzer.
 func NewQueryAnalyzer() *QueryAnalyzer {
-	return &QueryAnalyzer{}
+	cache, err := lru.New(256)
+	logger := logging.NewLogger("warn", "", "")
```

Comment:
We shouldn't initialize the logger here, we should pass it from the function parameter.

Reply:
ok